### PR TITLE
+order

### PR DESCRIPTION
**Pull request by Vosker**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public List<Order> getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession().getSession()) {
+            Query<Order> getAllOrdersQuery = session.createQuery("from Order o "
+                    + "left join fetch o.tickets "
+                    + "where o.user = :user", Order.class);
+            getAllOrdersQuery.setParameter("user", user);
+            return getAllOrdersQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get orders of " + user, e);
+        }
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java]: Do we need to catch an exception with the **_READ_** operations?

**Review by voloshynamaryna11 — COMMENTED** [review]
good job

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @Override
+    public List<Order> getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession().getSession()) {
+            Query<Order> getAllOrdersQuery = session.createQuery("from Order o "
+                    + "left join fetch o.tickets "
+                    + "where o.user = :user", Order.class);
+            getAllOrdersQuery.setParameter("user", user);
+            return getAllOrdersQuery.getResultList();
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java:46]: let's get only distinct values here

File: src/main/java/com/dev/cinema/Main.java [hunk 3]
```diff
@@ -53,5 +54,10 @@ public static void main(String[] args) throws AuthenticationException {
                 = (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
         shoppingCartService.addSession(movieSession, user1);
         shoppingCartService.addSession(movieSession, user1);
+
+        OrderService orderService
+                = (OrderService) injector.getInstance(OrderService.class);
+        orderService.completeOrder(shoppingCartService.getByUser(user1).getTickets(), user1);
+        orderService.getOrderHistory(user1).forEach(System.out::println);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java:61]: toString() is not overriden in Order

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**